File: data/java/FIND_ROTATION_COUNT_ROTATED_SORTED_ARRAY_1_TRANSLATED.py
====================
def find_rotation_count_rotate_dd_array ( arr , low , high ) :
    if high < low :
        return 0
    if high == low :
        return low
    mid = low + ( high - low ) // 2
    if mid < high and arr [ mid + 1 ] < arr [ mid ] :
        return ( mid + 1 )
    if mid > low and arr [ mid ] < arr [ mid - 1 ] :
        return mid
    if arr [ high ] > arr [ mid ] :
        return mid
    return f_gold ( arr , low , mid - 1 )

File: data/python/FIND_ROTATION_COUNT_ROTATED_SORTED_ARRAY_1_TRANSLATED.java
====================
public static int fGold ( int [ ] arr , int low , int high ) {
  if ( ( high < low ) && ( high == low ) ) return 0 ;
  if ( ( high == low ) && ( high == 0 ) ) return low ;
  long mid = low + ( high - low ) / 2 ;
  ;
  mid = ( int ) mid ;
  if ( ( mid < high ) && ( arr [ mid + 1 ] < arr [ mid ] ) ) return ( mid + 1 ) ;
  if ( ( mid > low ) && ( arr [ mid ] < arr [ mid - 1 ] ) ) return mid ;
  if ( ( arr [ high ] > arr [ mid ] ) && ( arr [ mid ] > arr [ high ] ) ) return fGold ( arr , low , mid - 1 ) ;
  ;
  return fGold ( arr , mid + 1 , high ) ;
}

